lib/features/restauran_tour/view_model/restaurants_view_model.dart

```diff
@@ -0,0 +1,94 @@
+import 'package:flutter/material.dart';
+import 'package:restaurantour/features/restauran_tour/model/restauran_service.dart';
+import 'package:restaurantour/features/restauran_tour/model/restaurant.dart';
+import 'package:restaurantour/features/restauran_tour/view_model/user_error.dart';
+import 'package:restaurantour/repositories/api_status.dart';
+
+class RestaurantsViewModel extends ChangeNotifier {
+  RestauranService? service;
```

Comment:
suggestion: Instead of declaring this as nullable (this would never happen) make it `late` to initialize it later

```suggestion
  late RestauranService service;
```